src/solvers/smt2/smt2_format.cpp

```diff
@@ -100,6 +100,25 @@ std::ostream &smt2_format_rec(std::ostream &out, const exprt &expr)
         << smt2_format(with_expr.where()) << ' '
         << smt2_format(with_expr.new_value()) << ')';
   }
+  else if(expr.id() == ID_array_list)
+  {
+    const auto &array_list_expr = to_multi_ary_expr(expr);
+
+    for(std::size_t i = 0; i < array_list_expr.operands().size(); i += 2)
+      out << "(store ";
```

Comment:
A recursive function would probably look nicer and make it easier to check that the output is well parenthesized.

Reply:
I was tempted to do that. However, we got bitten in the past by people building ridiculously large expressions, which then resulted in a stack overflow on Windows.